### PR TITLE
kafka-5118: Improve message for Kafka failed startup with non-Kafka data in data.dirs

### PR DESCRIPTION
**Pull request by huxihx**
Explicitly throwing clear exceptions when starting up a Kafka with some non-Kafka data in data.dirs.

### CODE REVIEW

**Comment by huxihx** [comment]
@junrao Please kindly review the PR. Thanks.

**Comment by huxihx** [comment]
@ijuma Could you kindly take some time to review this PR. Thanks.

**Comment by huxihx** [comment]
@ijuma Did you get any chances to review it. Thanks.

**Comment by huxihx** [comment]
@hachikuji Could you kindly take some time to review this PR? Thanks.

**Comment by huxihx** [comment]
@ijuma Hi, did you get any chances to review this PR?

**Comment by huxihx** [comment]
@guozhangwang Please kindly take some time to review this PR. Thanks.

**Comment by guozhangwang** [comment]
LGTM!

File: core/src/main/scala/kafka/log/Log.scala [hunk 1]
```diff
@@ -1362,14 +1362,17 @@ object Log {
...
+        "'" + dir.getName + "' is not in the form of topic-partition or " +
+        "ongoing-deleting directory(topic-partition.uniqueId-delete)\n" +
         "If a directory does not contain Kafka topic data it should not exist in Kafka's log " +
         "directory")
     }
 
     val dirName = dir.getName
     if (dirName == null || dirName.isEmpty || !dirName.contains('-'))
       throw exception(dir)
+    if (dirName.endsWith(DeleteDirSuffix) && !dirName.matches("^(\\S+)-(\\S+)\\.(\\S+)" + DeleteDirSuffix))
```
Comment (ijuma) [inline core/src/main/scala/kafka/log/Log.scala:1374]: We should probably compile the regex once instead of every time.

**Comment by guozhangwang** [comment]
Merged to trunk.

**Comment by guozhangwang** [comment]
@amethystic I changed your author signature from `amethystic <huxi_2b@hotmail.com>` to `Xi Hu <huxi_2b@hotmail.com>`, let me know if you prefer the former one, or you could also change your local profile accoordingly.

**Comment by huxihx** [comment]
@guozhangwang Thanks for the review and please keep my preferred name `amethystic` used. Appreciated.

**Pull request closed without merging**